trieste/acquisition/rule.py

```diff
@@ -200,14 +202,165 @@ def acquire(
 
         return points
 
 
+class AsyncEfficientGlobalOptimization(
+    AcquisitionRule[
+        State[Optional["AsyncEfficientGlobalOptimization.State"], TensorType], SP_contra
+    ]
+):
+    """AsyncEfficientGlobalOptimization rule, as name suggests,
+    is designed for asynchronous BO scenarios.
+    By asynchronous BO we understand a use case when multiple objective function
+    can be launched in parallel and are expected to arrive at different times.
+    Instead of waiting for the rest of observations to return, we want to immediately
+    use acquisition function to launch a new observation and avoid wasting computational resources.
+
+    To make the best decision about next point to observe, acquisition function
+    needs to be aware of currently running observations.
+    We call such points "pending". AsyncEGO provides a ``AsyncEfficientGlobalOptimization.State``
+    object that keeps track of pending points.
+    """
+
+    @dataclass(frozen=True)
+    class State:
+        pending_points: TensorType
+
+    def __init__(
+        self,
+        builder: GreedyAcquisitionFunctionBuilder | SingleModelGreedyAcquisitionBuilder,
+        optimizer: AcquisitionOptimizer[SP_contra] | None = None,
+        num_query_points: int = 1,
+    ):
+        """
+        :param builder: Acquisition function builder. Only greedy batch approaches are supported,
+            because they can be told what points are pending.
+        :param optimizer: The optimizer with which to optimize the acquisition function built by
+            ``builder``. This should *maximize* the acquisition function, and must be compatible
+            with the global search space. Defaults to
+            :func:`~trieste.acquisition.optimizer.automatic_optimizer_selector`.
+        :param num_query_points: The number of points to acquire.
+        """
+        if num_query_points <= 0:
+            raise ValueError(
+                f"Number of query points must be greater than 0, got {num_query_points}"
+            )
+
+        if builder is None:
+            raise ValueError("Please specify an acquisition builder")
+
+        if not isinstance(
+            builder, (GreedyAcquisitionFunctionBuilder, SingleModelGreedyAcquisitionBuilder)
+        ):
+            raise NotImplementedError(
+                f"""Only greedy acquisition strategies are supported
+                    by AsyncEGO, got {type(builder)}"""
+            )
+
+        if optimizer is None:
+            optimizer = automatic_optimizer_selector
+
+        if isinstance(builder, SingleModelGreedyAcquisitionBuilder):
+            builder = builder.using(OBJECTIVE)
+
+        self._builder: GreedyAcquisitionFunctionBuilder = builder
+        self._optimizer = optimizer
+        self._num_query_points = num_query_points
+        self._acquisition_function: Optional[AcquisitionFunction] = None
+
+    def __repr__(self) -> str:
+        """"""
+        return f"""AsyncEGO(
```

Comment:
perhaps AsyncEfficientGlobalOptimization rather than AsyncEGO?

Reply:
Yes, this should match the class name.